### PR TITLE
LCP: Test different scenarios of image upscaling

### PR DESCRIPTION
**Pull request by noamr**
Check that the actual scaled and intersected image size is considered, with both regular and background images.

Tests https://github.com/w3c/largest-contentful-paint/pull/99

### CODE REVIEW

File: largest-contentful-paint/image-upscaling.html [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                    resolve(entry.size);
+            });
+        }).observe({type: 'largest-contentful-paint'}));
+        popup.document.body.appendChild(container);
+        const timeout = new Promise(resolve => t.step_timeout(() => resolve('not reported'), 1000));
+        return {lcpSize: (await Promise.any([entryReported, timeout])), naturalSize};
+    }
+
+    const load_background_image_and_get_lcp_size = (t, style) =>
+        load_image_and_get_lcp_size(t, {display: 'none'},
```
Comment (npm1) [inline largest-contentful-paint/image-upscaling.html:41]: Why display:none?

File: largest-contentful-paint/image-upscaling.html [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    setup(() =>
+        assert_implements(window.LargestContentfulPaint, "LargestContentfulPaint is not implemented"));
+
+    const imageURL = `${window.location.origin}/images/blue.png`;
+    async function load_image_and_get_lcp_size(t, imageStyle = {}, containerStyle = {}) {
+        const popup = window.open('about:blank');
+        t.add_cleanup(() => popup.close());
+        const image = popup.document.createElement('img');
+        image.src = imageURL;
+        await image.decode();
```
Comment (npm1) [inline largest-contentful-paint/image-upscaling.html:19]: I feel this needs some comments to explain why we are creating so many different things

File: largest-contentful-paint/image-upscaling.html (hunk 1 above)
Comment (noamr) [reply largest-contentful-paint/image-upscaling.html:41]: To have only the div with the background and not the image.

File: largest-contentful-paint/image-upscaling.html (hunk 2 above)
Comment (noamr) [reply largest-contentful-paint/image-upscaling.html:21]: Added a couple of comments

**Review by npm1 — APPROVED** [review]

**Comment by noamr** [comment]
> This check doesn't work for `load_background_image_and_get_lcp_size` right? As the image container is not the `img`, but `div`, and the `id` of the `div is "".

You're right, can you post a fix to the test?

**Pull request merged**